app/controllers/application_controller.rb

```diff
@@ -9,6 +9,10 @@ def admin?
     user_signed_in? && current_user.admin?
   end
 
+  def json_api_token_auth?
```

Comment:
This would be better as an exclamation mark method, like `authenticate_admin!` below, so that it can be used as a `before_action`.